astropy/nddata/nduncertainty.py

```diff
@@ -378,3 +396,6 @@ def propagate_divide(self, other_nddata, result_data):
              result_data)
 
         return result_uncertainty
+
+uncertainty_type_dict = {item.uncertainty_type:item
+                         for item in NDUncertainty.__subclasses__()}
```

Comment:
@embray I think we are ditching this approach again (some parts of the PR will stay). Mainly because of concerns similar to yours.